specification/agrifood/resource-manager/Microsoft.AgFoodPlatform/preview/2023-06-01-preview/agfood.json

```diff
@@ -2718,6 +2718,10 @@
           "minLength": 1,
           "type": "string"
         },
+        "roleAssignmentId": {
+          "description": "Role Assignment Id.",
```

Comment:
Role Assignment id of what? What does it actually mean? Are you supposed to supply a role assignment id of either a custom or built-in role assignment that you've already created? Which tenant can it be in? What permissions will the role need?

Reply:
This field is just for internal purposes, for tracking the role-assignment-id created during an automated flow. Customer doesn't have to interact with this anywhere.

Reply:
Why are you exposing it to the customer view then?

Reply:
Since it's needed for portal api calls to get the relevant details which cannot be obtained otherwise

Reply:
Would you please share us more about how Portal will use this field?

Reply:
During deployment of a new solution, a guid for role-assignment-id is created and stored through arm-template of solution deployment.
When the installed solutions are fetched through GET solutions api-call, the roleAssignmentId value is also received in response and is used in the solution-delete flow where the linked role-assignment is deleted along with the solution.

Reply:
So similarly, user can make same deployment with rest api, right?
1. Create a role assignment
2. Call the PUT solution with the role assignment id
3. Call the GET solution to get the role assignment id
4. Delete the solution and delete the role assignment with its id

Reply:
Yes, but for UI driven experience, the appropriate value for role-assignment-id is difficult to find in portal.
This value is not directly visible to customer in portal, unless done via REST API